main.go

```diff
@@ -84,63 +86,62 @@ func main() {
 }
 
 func run(cfg Config, additionalSchemaPaths []string, updateDependencies bool) error {
-	return foreachChart(cfg.ChartsDirectory.path, func(base string) error {
-		logger := log.With().Str("chart", filepath.Base(base)).Logger()
-		valuesFiles, err := os.ReadDir(filepath.Join(base, TestsPath))
+	var validations_errors []error
 
+	err := filepath.WalkDir(cfg.ChartsDirectory.path, func(path string, dirent fs.DirEntry, err error) error {
+		logger := log.With().Str("path", path).Logger()
 		if err != nil {
-			logger.Error().Stack().Err(err).Msgf("Could not open directory %s", base)
-			return err
-		}
-
-		for _, file := range valuesFiles {
-			name := file.Name()
-			fileLogger := logger.With().Str("file", name).Logger()
-			fileLogger.Printf("Validating chart %s with values file %s...\n", filepath.Base(base), name)
-			manifests, err := runHelm(cfg.Helm.path, base, name, updateDependencies)
-
-			if err != nil {
-				fileLogger.Printf("Could not run Helm: %s\nStdout: %s\n", err, manifests.String())
-				return err
-			}
-
-			// to use kubeconform as a library would need us to practically
-			// reimplement its CLI
-			// <https://github.com/yannh/kubeconform/blob/dcc77ac3a39ed1fb538b54fab57bbe87d1ece490/cmd/kubeconform/main.go#L47>,
-			// so instead we shell out to it
-			output, err := runKubeconform(manifests, cfg.Kubeconform.path, cfg.Strict, additionalSchemaPaths, cfg.KubernetesVersion)
-
-			fileLogger.Info().Msgf("Output: %s", output)
+			logger.Warn().Err(err).Msg("skipping path (error)")
+
+		} else if dirent.IsDir() && regexp.MustCompile("^"+cfg.RegexSkipDir+"$").MatchString(dirent.Name()) {
```

Comment:
Is it possible to move the regexp out of the inner function and use [`QuoteMeta`](https://pkg.go.dev/regexp#QuoteMeta)?

Reply:
Moving the regex out, absolutely. Regarding QuoteMeta, though, isn't that going to prevent people to use this as a regex ? (I'm realising that `.git` should be `\.git` ...).
At first I was thinking to have a list of litteral string rather than a regex, quoting the regex would prevent doing things such as `.*` or `skip_this_one|and_this|and_this` right ?

Reply:
I completely forgot that it’s *meant* to be a regex! 🤦‍♂️ Sorry about that. Ignore me.

Reply:
I completely forgot that it’s *meant* to be a regex! 🤦‍♂️ Ignore me.